projects/dsp-ui/src/lib/search/fulltext-search/fulltext-search.component.spec.ts

```diff
@@ -0,0 +1,399 @@
+    const dspConnSpy = {
+        admin: {
+            projectsEndpoint: jasmine.createSpyObj('projectsEndpoint', ['getProjects', 'getProjectByIri'])
+        }
+    };
+
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            declarations: [
+                FulltextSearchComponent,
+                TestHostFulltextSearchComponent
+            ],
+            imports: [
+                OverlayModule,
+                RouterTestingModule,
+                FormsModule,
+                BrowserAnimationsModule,
+                MatMenuModule,
+                MatInputModule,
+                MatTooltipModule,
+                MatIconModule,
+                MatDividerModule,
+                MatListModule
+            ],
+            providers: [
+                {
+                    provide: DspApiConnectionToken,
+                    useValue: dspConnSpy
+                },
+                {
+                    provide: Router, useValue: mockRouter
+                }
+            ]
+        })
+            .compileComponents();
+    }));
+
+    beforeEach(() => {
+
+        const valuesSpy = TestBed.inject(DspApiConnectionToken);
+
+        (valuesSpy.admin.projectsEndpoint as jasmine.SpyObj<ProjectsEndpointAdmin>).getProjects.and.callFake(
+            () => {
+                const projects = MockProjects.mockProjects();
+                return of(projects);
+            }
+        );
+
+        testHostFixture = TestBed.createComponent(TestHostFulltextSearchComponent);
+        testHostComponent = testHostFixture.componentInstance;
+        testHostFixture.detectChanges();
+
+        hostCompDe = testHostFixture.debugElement;
+        fulltextSearchComponentDe = hostCompDe.query(By.directive(FulltextSearchComponent));
+    });
+
+    // Mock localStorage
+    beforeEach(() => {
+        let store = {};
+
+        spyOn(localStorage, 'getItem').and.callFake((key: string): string => {
+            return store[key] || null;
+        });
+        spyOn(localStorage, 'removeItem').and.callFake((key: string): void => {
+            delete store[key];
+        });
+        spyOn(localStorage, 'setItem').and.callFake((key: string, value: string): string => {
+            return store[key] = value as string;
+        });
+        spyOn(localStorage, 'clear').and.callFake(() => {
+            store = {};
+        });
+    });
+
+    it('should create', () => {
+        expect(testHostComponent).toBeTruthy();
+        expect(testHostComponent.fulltextSearch).toBeTruthy();
+    });
+
+    it('should get projects on init', () => {
+        const projSpy = TestBed.inject(DspApiConnectionToken);
+
+        expect(testHostComponent.fulltextSearch.projects).toBeDefined();
+        expect(testHostComponent.fulltextSearch.projects.length).toEqual(8);
+        expect(testHostComponent.fulltextSearch.projectfilter).toEqual(true);
+
+        // TO FIX: expect(projSpy.admin.projectsEndpoint.getProjects).toHaveBeenCalledTimes(1);
```

Comment:
@tobiasschweizer when I run this test, this expect fails because the method is called multiple times in a random way (sometimes it is called 1, 3, 6, 7 ,8 ,9, etc...). I do not understand the logic behind the error message. Maybe I have to add something in the beforeEach?

Reply:
When I run the test "'should do a search'", I get the following errors:

![Screenshot 2020-06-23 at 12 52 59](https://user-images.githubusercontent.com/6000023/85395498-887b1400-b550-11ea-96fc-b4f5a569b750.png)

I am not sure where the zone errors (XMLHTTPRequest) come from. The router or some other dependency that should be mocked? 

I think the `detach` problem comes from an uninitialised `this.overlayRef`. In this case, it is `undefined`.